src/test/java/BaseTest.java

```diff
@@ -0,0 +1,15 @@
+import org.openqa.selenium.WebDriver;
+import org.testng.Assert;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+public class BaseTest {
+    protected WebDriver driver = BaseDriver.initDriver();
+
+    BasePage basePage = new BasePage();
```

Comment:
In the future, you will have 1000 pages. Your BaseTest will contain many objects. Could you please add the needed pages in the test class where it will be used? And another thing, you should not use BasePage, please use child pages instead